### ISSUE TITLE
Add support for function in filter

### ISSUE THREAD

**Issue by omerlh** [ownaginatious/mixpanel-jql#4]
It would be nice to support function as a filter argument. Currently this is not working because the library automatically added the `return` at the start of the filter, but we could add overload that do not do this.

**Comment by ownaginatious** [comment]
That's a good idea. Would this seem like reasonable syntax to you while maintaining backwards compatibility?

```python
query = JQL(api_secret, params)\
          .filter('e.properties.B == 2')\
          .raw_filter(
              'function (e) {'
              '    return e.properties.x == 7'
              '}'
          )
         ...
```
I'll add support if so.

Also, sorry for the late response. GitHub never sent me a notification about this for some reason :/

**Comment by omerlh** [comment]
Yes, this looks pretty good - let you the option to do what ever you want.
Not sure about GitHub - guess this is something you could configure...

**Comment by ownaginatious** [comment]
I decided to try and make something more easily extensible:

```python
from mixpanel_jql import JQL, raw
...
query = JQL(api_secret, params)\
          .filter(raw(
              " function(e) {"
              "   if (e.x > 3) {"
              "     return true;"
              "   } else {"
              "     return false;"
              "   }"
              " )"
          ))\
          .filter('e.properties.F == "hello"')
```

This is now available in PyPI as version `0.3.post16`

**Comment by omerlh** [comment]
Thanks, that looks great!
